deps/wazuh_testing/wazuh_testing/tools/performance/statistic.py

```diff
@@ -40,44 +47,61 @@ class StatisticMonitor:
         dst_dir (str): directory to store the CSVs. Defaults to temp directory.
         csv_file (str): path to the CSV file.
         target (str): target file to monitor.
+        parse_json (bool): Determine if the file is a JSON file. Default False.
     """
 
-    def __init__(self, target='agent', time_step=5, dst_dir=gettempdir()):
+    def __init__(self, target='agent', time_step=5, dst_dir=gettempdir(), use_state_file=False, ip = 'localhost', port = '55000'):
```

Comment:
IMHO It could be confusing including this logic in the constructor.

For logcollector and agent which that are not supported by the daemon endpoint, we should raise a failure in case of not enabling the use of state file

It should be the statistic script that will handle which daemon is supported by the API, States or both